src/uid2StorageManager.ts

```diff
@@ -0,0 +1,47 @@
+import { UID2 } from "./uid2Sdk";
+import { isValidIdentity, Uid2Identity } from "./Uid2Identity";
+import { UID2CookieManager } from "./uid2CookieManager";
+import { UID2LocalStorageManager } from "./uid2LocalStorageManager";
+import { Uid2Options } from "./Uid2Options";
+
+export class UID2StorageManager {
+  private _cookieManager: UID2CookieManager | undefined;
+  private _localStorageManager: UID2LocalStorageManager | undefined;
+
+  private _opts: Uid2Options;
+  constructor(opts: Uid2Options) {
+    this._opts = opts;
+    this._cookieManager = new UID2CookieManager({ ...opts });
+    this._localStorageManager = new UID2LocalStorageManager();
+  }
+
+  public loadIdentityFromPreferredStorageWithFallback(): Uid2Identity | null {
+    const localStorageIdentity = this._localStorageManager?.loadIdentityFromLocalStorage();
+    const cookieIdentity = this._cookieManager?.loadIdentityFromCookie();
+    const shouldUseCookie = cookieIdentity && (!localStorageIdentity || cookieIdentity.identity_expires > localStorageIdentity.identity_expires);
+    return shouldUseCookie ? cookieIdentity : localStorageIdentity ?? null;
+  };
+
+  public loadIdentity(): Uid2Identity | null {
+    return this._opts.useCookie
+      ? this._cookieManager?.loadIdentityFromCookie() ?? null
+      : this._localStorageManager?.loadIdentityFromLocalStorage() ?? null;
+  }
+
+  public setValue(identity: Uid2Identity) {
+    if (this._opts.useCookie) {
+      this._cookieManager?.setCookie(identity);
+    }
+    else if (this._opts.useCookie === false) {
+      this._localStorageManager?.setValue(identity);
+      if (this._localStorageManager?.loadIdentityFromLocalStorage()) this._cookieManager?.removeCookie();
```

Comment:
As `this._localStorageManager?.setValue(identity);` is called in the last line, it is possible this returns false? If so, is this equivalent to `if(identity)` ? 

Nit: maybe use return in the first if condition, then we don't need elseif and else?
```
    if (this._opts.useCookie) {
      this._cookieManager?.setCookie(identity);
      return;
    }

    this._localStorageManager?.setValue(identity);
    if (
      this._opts.useCookie === false &&
      this._localStorageManager?.loadIdentityFromLocalStorage()
    ) {
      this._cookieManager?.removeCookie();
    }
```

Reply:
> As this._localStorageManager?.setValue(identity); is called in the last line, it is possible this returns false? If so, is this equivalent to if(identity) ?
I think it is possible it returns falsy if this._localStorageManager is somehow undefined or if setValue does not work for some reason. This is just extra defensive code to address this comment from Lionell: https://github.com/IABTechLab/uid2-web-integrations/pull/21#discussion_r1321003036

Reply:
Got it